diff --git a/src/decode_r2007.c b/src/decode_r2007.c
index 60cf126b..52993f8e 100644
--- a/src/decode_r2007.c
+++ b/src/decode_r2007.c
@@ -48,6 +48,7 @@ static unsigned int cur_ver = 0;
 // should be a bit larger then the filesize.
 #define DBG_MAX_COUNT 0x100000
 #define DBG_MAX_SIZE 0xff0000 /* should be dat->size */
+#define MAX_R2007_SECTION_DECOMP_SIZE UINT64_C (0x2f000000) /* 790 MB */

 /* page map */
 typedef struct _r2007_page
@@ -807,7 +808,7 @@ read_data_section (Bit_Chain *sec_dat, Bit_Chain *dat,
     }

   max_decomp_size = section->data_size;
-  if (max_decomp_size > 0x2f000000) // 790Mb
+  if (max_decomp_size > MAX_R2007_SECTION_DECOMP_SIZE)
     {
       LOG_ERROR ("Invalid max decompression size %" PRIu64, max_decomp_size);
       return DWG_ERR_INVALIDDWG;
@@ -940,7 +941,7 @@ read_sections_map (Bit_Chain *dat, int64_t size_comp, int64_t size_uncomp,
       // debugging sanity
 #if 1
       /* compressed */
-      if (section->data_size > 10 * dat->size
+      if (section->data_size > MAX_R2007_SECTION_DECOMP_SIZE
           || section->name_length >= (int64_t)dat->size
           || section->name_length >= 48)
         {
